packages/chatbot-server-mongodb-public/src/config.eval.ts

```diff
@@ -0,0 +1,207 @@
+import { AnswerRelevancy, ContextRelevancy, Faithfulness } from "autoevals";
+import fs from "fs";
+import path from "path";
+import { getConversationsEvalCasesFromYaml } from "./test/getConversationEvalCasesFromYaml";
+import { strict as assert } from "assert";
+
+interface ConversationEvalCaseInput {
+  previousConversation: Conversation;
+  latestMessageText: string;
+}
+interface ConversationEvalCase
+  extends EvalCase<ConversationEvalCaseInput, unknown, unknown> {
+  name: string;
+  input: ConversationEvalCaseInput;
+  tags?: MongoDbTag[];
+}
+
+interface ConversationTaskOutput {
+  assistantMessageContent: string;
+  context?: string[];
+  allowedQuery: boolean;
+}
+
+type ConversationEvalScorer = EvalScorer<
+  ConversationEvalCaseInput,
+  ConversationTaskOutput,
+  unknown
+>;
+
+// -- Evaluation metrics --
+const RetrievedContext: ConversationEvalScorer = async (args) => {
+  args.output.context;
+  return {
+    name: "RetrievedContext",
+    score: args.output.context?.length ? 1 : 0,
+  };
+};
+
+const AllowedQuery: ConversationEvalScorer = async (args) => {
+  return {
+    name: "AllowedQuery",
+    score: args.output.allowedQuery ? 1 : 0,
+  };
+};
+
+function getConversationRagasConfig(
+  scorerArgs: Parameters<ConversationEvalScorer>[0],
+  judgeModelConfig: {
+    model: string;
+    azureOpenAi: {
+      apiKey: string;
+      apiVersion: string;
+      endpoint: string;
+    };
+  }
+) {
+  return {
+    output: scorerArgs.output.assistantMessageContent,
+    context: scorerArgs.output.context,
+    input: scorerArgs.input.latestMessageText,
+    model: judgeModelConfig.model,
+    azureOpenAi: judgeModelConfig.azureOpenAi,
+  };
+}
+
+const judgeModelConfig = {
+  model: JUDGE_LLM,
+  embeddingModel: JUDGE_EMBEDDING_MODEL,
+  azureOpenAi: {
+    apiKey: OPENAI_API_KEY,
+    apiVersion: OPENAI_API_VERSION,
+    endpoint: OPENAI_ENDPOINT,
+  },
+};
+
+const ConversationFaithfulness: ConversationEvalScorer = async (args) => {
+  return Faithfulness(getConversationRagasConfig(args, judgeModelConfig));
+};
+
+// Commenting out until this PR is merged and released https://github.com/braintrustdata/autoevals/pull/94
+// const ConversationAnswerRelevancy: ConversationEvalScorer = async (args) => {
+//   return AnswerRelevancy({
+//     ...getConversationRagasConfig(args, judgeModelConfig),
+//     embeddingModel: JUDGE_EMBEDDING_MODEL,
+//   });
+// };
+
+const ConversationContextRelevancy: ConversationEvalScorer = async (args) => {
+  return ContextRelevancy(getConversationRagasConfig(args, judgeModelConfig));
+};
+
+Eval("mongodb-chatbot-conversations", {
+  data: async () => {
+    const basePath = path.resolve(__dirname, "..", "evalCases");
+    const miscCases = getConversationsEvalCasesFromYaml(
+      fs.readFileSync(path.resolve(basePath, "conversations.yml"), "utf8")
+    );
+    const faqCases = getConversationsEvalCasesFromYaml(
+      fs.readFileSync(path.resolve(basePath, "faq_conversations.yml"), "utf8")
+    );
```

Comment:
When do you see us putting a set of evaluation cases in this file vs the `processors/*.eval.ts` approach?

I'm a bit confused on the name of the file too. Could we call this e.g. `conversations.eval.ts` or is there something particular about `config.ts` that we're evaluating here?

Reply:
> When do you see us putting a set of evaluation cases in this file vs the `processors/*.eval.ts` approach?

i see this file as our "end-to-end" evaluation. 

in contrast, the `processors/*.eval.ts` are for each AI component. i see these as more like "unit" evaluations.

> I'm a bit confused on the name of the file too. Could we call this e.g. `conversations.eval.ts` or is there something particular about `config.ts` that we're evaluating here?

we can def change the name. i was also not loving the name. just put this as it is b/c it's largely evaluating the stuff in the `config.ts` file. i think `conversations.eval.ts` makes sense here.

---

more philosophically, i think there's a larger conversation that we ought to have about:

1. how to programmatically run the evals
2. how to structure evals for different features. for example for the OM/CM stuff your doing, i think we agree that we shoudl accompany that work w/ some evals. but where do we put them? in this `conversations.evals.ts` file/evaluation? create a separate file, etc

Reply:
my $.02 is put them in the same yaml files and include tags for the relevant products, run the whole eval suite in the `conversations.evals.ts` file. see how we perform on the new eval cases and make sure there's no regression elsewhere.